File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/Sincronizacao.java
package org.jisho.textosJapones.model.entities;

import org.jisho.textosJapones.model.enums.Conexao;

import java.time.LocalDateTime;

public class Sincronizacao {

    private Conexao conexao;
    private LocalDateTime envio;
    private LocalDateTime recebimento;

    public Conexao getConexao() {
        return conexao;
    }

    public LocalDateTime getEnvio() {
        return envio;
    }

    public void setEnvio(LocalDateTime envio) {
        this.envio = envio;
    }

    public LocalDateTime getRecebimento() {
        return recebimento;
    }

    public void setRecebimento(LocalDateTime recebimento) {
        this.recebimento = recebimento;
    }

    public Sincronizacao(Conexao conexao, LocalDateTime envio, LocalDateTime recebimento) {
        this.conexao = conexao;
        this.envio = envio;
        this.recebimento = recebimento;
    }
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/processatexto/Sincronizacao.kt
package br.com.fenix.processatexto.model.entities.processatexto

import br.com.fenix.processatexto.model.entities.EntityBase
import br.com.fenix.processatexto.model.enums.Conexao
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.Id
import java.time.LocalDateTime


@Entity
data class Sincronizacao(
    @Id
    @Enumerated(EnumType.STRING)
    var conexao: Conexao = Conexao.PROCESSA_TEXTO,
    var envio: LocalDateTime = LocalDateTime.now(),
    var recebimento: LocalDateTime = LocalDateTime.now()
) : EntityBase<Conexao, Sincronizacao>() {

    constructor() : this(Conexao.PROCESSA_TEXTO) {}

    override fun getId(): Conexao = conexao
    override fun setId(id : Conexao?) { }

    override fun create(conexao: Conexao): Sincronizacao = Sincronizacao(conexao)

}